File: Patient.R
# Importing libraries for Survival Analysis
library(survminer)
library(survival)

# Loading the Patient Dataset
Patient <- read.csv(file.choose())
View(Patient)

attach(Patient)
str(Patient)


# Define variables 
time <- Followup
event <- Eventtype  # Event type can take 2 values 0 or 1

# Descriptive statistics
summary(time)
table(time)
table(event)

# Kaplan-Meier non-parametric analysis
kmsurvival <- survfit(Surv(time,event) ~ 1)
summary(kmsurvival)

plot(kmsurvival, xlab="Time", ylab="Survival Probability")
ggsurvplot(kmsurvival, data=Patient, risk.table = TRUE)


# Kaplan-Meier non-parametric analysis by group
kmsurvival1 <- survfit(Surv(time, event) ~ PatientID)
summary(kmsurvival1)

plot(kmsurvival1, xlab="Follow UP", ylab="Survival Probability")
ggsurvplot(kmsurvival1, data=Patient, risk.table = TRUE)

File: Patient.py
# Importing Required Packages
import pandas as pd
from lifelines import KaplanMeierFitter # to fit the survival analysis


# Loading the the Patients data
Patient = pd.read_csv("C:/Users/SHAJIUDDIN MOHAMMED/Desktop/Module-Survival Analysis/Patient.csv")
Patient.head()
Patient.describe()

Patient["Followup"].describe()

# Followuo is referring to time 
F = Patient.Followup

# Initiating the KaplanMeierFitter model
kmf = KaplanMeierFitter()

# Fitting KaplanMeierFitter model on Time and Events for life
kmf.fit(F, event_observed=Patient.Eventtype)

# Time-line estimations plot 
kmf.plot()
